### PR TITLE
Add trivy deployment

### PR DESCRIPTION
**Pull request by nd4pa**
There are still ongoing issue with the image, config seems to be exposed correctly to the pod but the binary does not seem to read it properly, need some further testing

Signed-off-by: Antoine Blondeau <antoine@blondeau.me>

### CODE REVIEW

File: apis/goharbor.io/v1alpha2/trivy_types.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	// The URL of the HTTP proxy server
+	HTTProxy string `json:"httpProxy,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Pattern="https?://.+"
+	// The URL of the HTTPS proxy server
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// The URLs that the proxy settings do not apply to
```
Comment (XavierDuthil) [inline apis/goharbor.io/v1alpha2/trivy_types.go:142]: Is it comma-separated like the other lists?

File: apis/goharbor.io/v1alpha2/trivy_types.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+	// Trivy reports directory
+	ReportsDir string `json:"reportsDir,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// The flag to enable or disable Trivy debug mode
+	DebugMode bool `json:"debugMode,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// TODO +kubebuilder:validation:Enum={"os","library"}
```
Comment (XavierDuthil) [inline apis/goharbor.io/v1alpha2/trivy_types.go:99]: Remove TODO?

File: apis/goharbor.io/v1alpha2/trivy_types.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+	// +kubebuilder:default={"os","library"}
+	// Comma-separated list of vulnerability types.
+	// Possible values are os and library.
+	VulnType []string `json:"vulnType,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// TODO +kubebuilder:validation:Enum={"UNKNOWN","LOW","MEDIUM","HIGH","CRITICAL"}
```
Comment (XavierDuthil) [inline apis/goharbor.io/v1alpha2/trivy_types.go:106]: Remove TODO?

File: apis/goharbor.io/v1alpha2/trivy_types.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+	CacheDir string `json:"cacheDir,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="/home/scanner/.cache/reports"
+	// Trivy reports directory
+	ReportsDir string `json:"reportsDir,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// The flag to enable or disable Trivy debug mode
```
Comment (XavierDuthil) [inline apis/goharbor.io/v1alpha2/trivy_types.go:95]:
```suggestion
	// Enable Trivy debug mode
```

File: apis/goharbor.io/v1alpha2/trivy_types.go [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+	// +kubebuilder:validation:Optional
+	// TODO +kubebuilder:validation:Enum={"UNKNOWN","LOW","MEDIUM","HIGH","CRITICAL"}
+	// +kubebuilder:default={"UNKNOWN","LOW","MEDIUM","HIGH","CRITICAL"}
+	// Comma-separated list of vulnerabilities
+	// severities to be displayed
+	Severity []string `json:"severity,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// The flag to display only fixed vulnerabilities
```
Comment (XavierDuthil) [inline apis/goharbor.io/v1alpha2/trivy_types.go:114]:
```suggestion
	// Display only fixed vulnerabilities
```

File: apis/goharbor.io/v1alpha2/trivy_types.go [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+	IgnoreUnfixed bool `json:"ignoreUnfixed,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// The flag to enable or disable Trivy DB downloads from GitHub
```
Comment (XavierDuthil) [inline apis/goharbor.io/v1alpha2/trivy_types.go:119]:
```suggestion
	// Disable Trivy DB downloads from GitHub
```

File: apis/goharbor.io/v1alpha2/trivy_types.go [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+	SkipUpdate bool `json:"skipUpdate,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// The GitHub access token to download Trivy DB (see GitHub rate limiting)
+	GithubToken string `json:"githubToken,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// The flag to skip verifying registry certificate
```
Comment (XavierDuthil) [inline apis/goharbor.io/v1alpha2/trivy_types.go:128]:
```suggestion
	// Skip verification of registry certificate
```

File: apis/goharbor.io/v1alpha2/trivy_types.go (hunk 1 above)
Comment (XavierDuthil) [reply apis/goharbor.io/v1alpha2/trivy_types.go:142]: It's OK - comment taken directly from trivy's repository.

File: apis/goharbor.io/v1alpha2/trivy_types.go (hunk 2 above)
Comment (XavierDuthil) [reply apis/goharbor.io/v1alpha2/trivy_types.go:99]: It's OK - comment taken directly from trivy's repository.

File: apis/goharbor.io/v1alpha2/trivy_types.go (hunk 3 above)
Comment (XavierDuthil) [reply apis/goharbor.io/v1alpha2/trivy_types.go:106]: It's OK - comment taken directly from trivy's repository.

File: apis/goharbor.io/v1alpha2/trivy_types.go (hunk 4 above)
Comment (XavierDuthil) [reply apis/goharbor.io/v1alpha2/trivy_types.go:95]: It's OK - comment taken directly from trivy's repository.

File: apis/goharbor.io/v1alpha2/trivy_types.go (hunk 5 above)
Comment (XavierDuthil) [reply apis/goharbor.io/v1alpha2/trivy_types.go:114]: It's OK - comment taken directly from trivy's repository.

File: apis/goharbor.io/v1alpha2/trivy_types.go (hunk 6 above)
Comment (XavierDuthil) [reply apis/goharbor.io/v1alpha2/trivy_types.go:119]: It's OK - comment taken directly from trivy's repository.

File: apis/goharbor.io/v1alpha2/trivy_types.go (hunk 7 above)
Comment (XavierDuthil) [reply apis/goharbor.io/v1alpha2/trivy_types.go:128]: It's OK - comment taken directly from trivy's repository.

**Comment by jMonsinjon** [comment]
This pull request has been replaced by this one:
https://github.com/goharbor/harbor-operator/pull/70

**Pull request closed without merging**